usaspending_api/search/v2/views/search.py

```diff
@@ -746,58 +746,43 @@ def __eq__(self, other):
-        titles matching the provided search text"""
-        json_request = request.data
-        fields = json_request.get("fields", None)
-        filters = json_request.get("filters", None)
-        order = json_request.get("order", "desc")
-        limit = json_request.get("limit", 10)
-        page = json_request.get("page", 1)
-        sort = json_request.get("sort", "Transaction Amount")
-
-        lower_limit = (page - 1) * limit
-        upper_limit = page * limit
 
-        if fields is None:
-            raise InvalidParameterException("Missing one or more required request parameters: fields")
-        elif len(fields) == 0:
-            raise InvalidParameterException("Please provide a field in the fields request parameter.")
-        if filters is None:
-            raise InvalidParameterException("Missing one or more required request parameters: filters")
-        if "award_type_codes" not in filters:
-            raise InvalidParameterException(
-                "Missing one or more required request parameters: filters['award_type_codes']")
-
-        if order not in ["asc", "desc"]:
-            raise InvalidParameterException("Invalid value for order: {}".format(order))
-        if sort not in fields:
-            raise InvalidParameterException("Sort value not found in fields: {}".format(sort))
-
-        filters['keyword'] = preprocess(filters['keyword'])
-        success, response, total, award_type = search_transactions(filters, fields, sort, order, lower_limit, limit)
+        models = [
+            {'name': 'fields', 'key': 'fields', 'type': 'array', 'array_type': 'text', 'text_type': 'search'},
+        ]
+        models.extend(AWARD_FILTER)
+        models.extend(PAGINATION)
+        for m in models:
+            if m['name'] in ('keyword', 'award_type_codes', 'sort'):
+                m['optional'] = False
+        validated_payload = TinyShield(models).block(request.data)
+
+        if validated_payload['sort'] not in validated_payload['fields']:
+            raise InvalidParameterException("Sort value not found in fields: {}".format(validated_payload['sort']))
+
+        lower_limit = (validated_payload['page'] - 1) * validated_payload['limit']
+        success, response, total = search_transactions(validated_payload, lower_limit, validated_payload['limit'] + 1)
         if not success:
             raise InvalidParameterException(response)
-        has_next = total > upper_limit
+
+        metadata = get_simple_pagination_metadata(len(response), validated_payload['limit'], validated_payload['page'])
+
         results = []
-        for transaction in response:
+        for transaction in response[:validated_payload['limit']]:
             transaction["internal_id"] = transaction["Award ID"]
             if 'display_award_id' in transaction:
                 transaction["Award ID"] = transaction['display_award_id']
                 del transaction['display_award_id']
-            elif award_type != 'contracts':
```

Comment:
Why are these being commented out?

Reply:
We no longer need to pull from different fields to get the "Award ID".

I commented out instead of removed because there is still a need to fix the Award ID sorting to use this new field (currently the sorting ordering for this column doesn't make sense to a human). Comment can be removed once that is fixed